docs/operating/extending.md

```diff
@@ -2,17 +2,61 @@
 
 IronFunctions is extensible so you can add custom functionality and extend the project without needing to modify the core.
 
+### Replaceable Components
+
+TODO
+
 ## Listeners
 
 Listeners are the main way to extend IronFunctions. 
 
-To add listeners, copy `main.go` into your own repo and add your own listener implementations. When ready, 
-compile your main package to create your extended version of IronFunctions.
+You can easily use listeners basically creating a struct with [valid methods](#Listener Requirements) and adding it to the `IronFunctions API`.
+
+Example:
+
+```
+package main
+
+import (
+    "context"
+
+    "github.com/iron-io/functions/api/server"
+    "github.com/iron-io/functions/api/models"
+)
+
+type myCustomListener struct{}
+
+func (c *myCustomListener) BeforeAppCreate(ctx context.Context, app *models.App) error { return nil }
+func (c *myCustomListener) AfterAppCreate(ctx context.Context, app *models.App) error { return nil }
 
-### AppListener
+func (c *myCustomListener) BeforeAppUpdate(ctx context.Context, app *models.App) error { return nil }
+func (c *myCustomListener) AfterAppUpdate(ctx context.Context, app *models.App) error { return nil }
 
-Implement `ifaces/AppListener` interface, then add it using:
+func (c *myCustomListener) BeforeAppDelete(ctx context.Context, appName string) error { return nil }
+func (c *myCustomListener) BeforeAppDelete(ctx context.Context, appName string) error { return nil }
 
-```go
-server.AddAppListener(myAppListener)
+function main () {
+    srv := server.New(/* Here all required parameters to initialize the server */)
+
+    // To add a listeners that listen to app's creation
+    srv.AddAppCreateListener(myCustomListener)
+
+    // To add listeners that listen to app's update
+    srv.AddAppUpdateListener(myCustomListener)
+
+    // To add listeners that listen to app's deletion
+    srv.AddAppDeleteListener(myCustomListener)
+
+    srv.Run()
+}
 ```
+
+#### Listener Requirements
+
+To be a valid listener your struct should respect interfaces combined or alone found in the file [listeners.go](/iron-io/functions/blob/master/api/ifaces/listeners.go)
+
+These are all available listeners:
+
+- AppCreateListener
```

Comment:
What do you think of adding a diagram or perhaps longe explanation for these interfaces? So people understand better exactly which part of the request cycle is being affected.